spec/models/assignment_spec.rb

```diff
@@ -61,28 +62,35 @@
       is_expected.to validate_numericality_of(:group_max).is_greater_than(0)
     end
 
-    it 'should create a valid assignment' do
-      assignment = create(:assignment)
-      expect(assignment).to be_valid
-    end
+    describe 'Validation of basic infos of an assignment' do
+      before :each do
+        @assignment = create(:assignment)
+      end
+      it 'should create a valid assignment' do
+        expect(@assignment).to be_valid
+      end
 
-    it 'should require case sensitive unique value for short_identifier' do
-      assignment = create(:assignment)
-      expect(assignment).to validate_uniqueness_of(:short_identifier)
-    end
-    it 'should have a nil parent_assignment by default' do
-      assignment = create(:assignment)
-      expect(assignment.parent_assignment).to be_nil
-    end
-    it 'should have a nil peer_review by default' do
-      assignment = create(:assignment)
-      expect(assignment.pr_assignment).to be_nil
+      it 'should require case sensitive unique value for short_identifier' do
+        expect(@assignment).to validate_uniqueness_of(:short_identifier)
+      end
+      it 'should have a nil parent_assignment by default' do
+        expect(@assignment.parent_assignment).to be_nil
+      end
+      it 'should have a nil peer_review by default' do
+        expect(@assignment.pr_assignment).to be_nil
+      end
+      it 'should not be a peer review if there is no parent_assignment_id' do
+        expect(@assignment.parent_assignment_id).to be_nil
+        expect(@assignment.is_peer_review?).to be false
+      end
     end
-    it 'should not be a peer review if there is no parent_assignment_id' do
-      assignment = create(:assignment)
-      expect(assignment.parent_assignment_id).to be_nil
-      expect(assignment.is_peer_review?).to be false
+
+    it 'should catch an invalid date' do
+      assignment = create(:assignment, due_date: '2020/02/31')  #31st day of february
+      !expect(assignment).to be_valid
```

Comment:
You should use the `not_to` method to get the inverse expectation for your test:

```
expect(assignment).not_to be_valid
```

see [here](http://rspec.info/documentation/3.6/rspec-expectations/RSpec/Matchers.html) for more examples